packages/astro/src/actions/runtime/route.ts

```diff
@@ -0,0 +1,38 @@
+import type { APIRoute } from '../../@types/astro.js';
+import { ApiContextStorage } from './store.js';
+import { formContentTypes, getAction } from './utils.js';
+import { ActionError } from './virtual/shared.js';
+
+export const POST: APIRoute = async (context) => {
+	const { request, url } = context;
+	const actionPathKeys = url.pathname.replace('/_actions/', '').split('.');
+	const action = await getAction(actionPathKeys);
+	const contentType = request.headers.get('Content-Type');
+	let args: unknown;
+	if (formContentTypes.some((f) => contentType?.startsWith(f))) {
+		args = await request.clone().formData();
+	} else if (contentType === 'application/json') {
+		args = await request.clone().json();
+	} else {
+		return new Response(null, { status: 415 });
+	}
+	let result: unknown;
+	try {
+		result = await ApiContextStorage.run(context, () => action(args));
+	} catch (e) {
+		if (e instanceof ActionError) {
+			return new Response(JSON.stringify(e), {
```

Comment:
Does this include the stack trace? Presumably you don't want that included in the client.

Reply:
Good catch. I'll ensure stack traces are dev-only. I noticed that's the practice tRPC uses